tutorials/haskell/bdd-in-haskell-with-hspec/code/src/Luhn/Internal.hs

```diff
@@ -0,0 +1,27 @@
+module Luhn.Internal
+  ( toDigits
+  , fromDigits
+  , doubleEveryOther
+  , sumDigits
+  )
+  where
+
+-- | Returns number digits as a list
+toDigits :: Integer -> [Integer]
+toDigits = go []
+  where go xs n
+          | n <= 0    = xs
+          | otherwise = go (m:xs) d where (d,m) = divMod n 10
+
+-- | Returns a number given its digits in a list
+fromDigits :: [Integer] -> Integer
+fromDigits = sum . zipWith (*) [10 ^ i | i <- [0..]] . reverse
+
+-- | Double every other digit on the list starting from the right most number
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOther = reverse . zipWith (*) (cycle [1,2]) . reverse
```

Comment:
@juank-pa This is not the solution included in the tutorial. Are you going to update it?